cmd/erasure-object.go

```diff
@@ -1034,11 +1034,9 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 	if goi.VersionID != "" {
 		markDelete = true
 	}
-	// Default deleteMarker to true if object is under versioning
+	// Default deleteMarker to true
 	deleteMarker := true
-	if gerr == nil {
-		deleteMarker = goi.VersionID != ""
-	}
+
```

Comment:
From what it looks like it's needed and has some niche requirement such as situations where `deleteMarker` is not created in situations when `gerr` returns `success` and `VersionID` is not specified. 

Specifically for non-versioned bucket I think it is suitable @poornas

Reply:
We should add the code back.